hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/KvMetadataJavaResolver.java

```diff
@@ -229,11 +232,17 @@ private KvMetadata resolveObjectMetadata(
     }
 
     private Class<?> loadClass(boolean isKey, Map<String, String> options) {
+        String formatProperty = options.get(isKey ? OPTION_KEY_FORMAT : OPTION_VALUE_FORMAT);
         String classNameProperty = isKey ? OPTION_KEY_CLASS : OPTION_VALUE_CLASS;
-        String className = options.get(classNameProperty);
 
-        if (className == null) {
-            throw QueryException.error("Unable to resolve table metadata. Missing '" + classNameProperty + "' option");
+        String className;
+        if (JAVA_FORMAT.equals(formatProperty)) {
+            className = options.get(classNameProperty);
+            if (className == null) {
+                throw QueryException.error("Unable to resolve table metadata. Missing '" + classNameProperty + "' option");
+            }
+        } else {
+            className = JavaClassNameResolver.resolveClassName(formatProperty);
```

Comment:
We can end up with null `className` here.

Reply:
Technically, we can't but we can defensively leave the check for the posterity.

Reply:
Then we can `assert` it.